api/post.go

```diff
@@ -233,7 +235,8 @@ func getPosts(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	etagChan := app.Srv.Store.Post().GetEtag(id, true)
 
-	if !HasPermissionToChannelContext(c, id, model.PERMISSION_CREATE_POST) {
+	if !app.SessionHasPermissionToChannel(c.Session, id, model.PERMISSION_EDIT_POST) {
```

Comment:
Hmmm I don't think it matters since you'll have both of those permissions at the same time but I'll switch it back to be sure